src/hotspot/share/oops/compressedKlass.cpp

```diff
@@ -32,9 +32,30 @@
 address CompressedKlassPointers::_base = nullptr;
 int CompressedKlassPointers::_shift = 0;
 size_t CompressedKlassPointers::_range = 0;
+uint64_t CompressedKlassPointers::_combo = 0;
 
 #ifdef _LP64
 
+void CompressedKlassPointers::set_base_and_shift(address thebase, int theshift) {
+  assert(UseCompressedClassPointers, "Why are we here?");
+  assert(theshift == 0 || theshift == LogKlassAlignmentInBytes, "invalid shift for klass ptrs");
+  _base = thebase;
+  _shift = theshift;
+
+  // we keep a composite word, `_combo`, containing base+shift+UseCCP, to load
+  // all three information with a single 64-bit load.
```

Comment:
```suggestion
  // Encode all three base+shift+UseCCP into a single 64-bit word.
  // This would allow optimizing the fast-path with a single load.
```